src/nonstandard.jl

```diff
@@ -56,3 +56,22 @@ constructorof(::Type{<:LinRange}) = linrange_constructor
 ### Expr: args get splatted
 # ::Expr annotation is to make it type-stable on Julia 1.3-
 constructorof(::Type{<:Expr}) = (head, args) -> Expr(head, args...)::Expr
+
+### Cholesky
+setproperties(C::LinearAlgebra.Cholesky, patch::NamedTuple{()}) = C
+function setproperties(C::LinearAlgebra.Cholesky, patch::NamedTuple{(:L,)})
+    return LinearAlgebra.Cholesky(C.uplo === 'U' ? permutedims(patch.L) : patch.L, C.uplo, C.info)
+end
+function setproperties(C::LinearAlgebra.Cholesky, patch::NamedTuple{(:U,)})
+    return LinearAlgebra.Cholesky(C.uplo === 'L' ? permutedims(patch.U) : patch.U, C.uplo, C.info)
+end
+function setproperties(C::LinearAlgebra.Cholesky, patch::NamedTuple{(:UL,)})
+    return LinearAlgebra.Cholesky(patch.UL, C.uplo, C.info)
+end
+function setproperties(C::LinearAlgebra.Cholesky, patch::NamedTuple)
```

Comment:
Better now @devmotion ?

Also, had to remove the `@nospecialize` as otherwise the default implementation will be hit